pkg/adapter/adapter.go

```diff
@@ -74,7 +74,7 @@ func NewABCIExecutor(
 
 	// Create a new Store with ABCI prefix
 	abciStore := NewStore(store)
-	rollkitStore := rstore.New(abciStore)
+	rollkitStore := rstore.New(store)
```

Comment:
This line was the key problem. instead of the original store, the prefix store was used.